external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -1374,4 +1377,185 @@ class AvroSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
       |}
     """.stripMargin)
   }
+
+  test("generic record converts to row and back") {
+    val nested =
+      SchemaBuilder.record("simple_record").fields()
+        .name("nested1").`type`("int").withDefault(0)
+        .name("nested2").`type`("string").withDefault("string").endRecord()
+    val mapDefault = new java.util.HashMap[String, String]()
+    mapDefault.put("a", "A")
+    val schema = SchemaBuilder.record("record").fields()
```

Comment:
We can use a json string to present the schema.

Reply:
Thanks, done in 9ee695c.